### DISCUSSION TITLE
How to Install on Laravel Vapor?

### DISCUSSION THREAD

**Discussion by saywebsolutions** [area17/twill#1065] [Q&A]
I get the following error when running:

```
vapor command cms-twill --command="php artisan twill:install"
```

Output:

```
==> Executing Function...

Status Code: 1

Output:

In Filesystem.php line 187:

file_put_contents(/var/task/routes/admin.php): Failed to open stream: Read-
only file system
```

**Comment by ifox** [comment]
Hi @saywebsolutions!

I would suggest installing Twill from your local computer or a server where you do have write access to the filesystem. Twill is trying to generate files for you so you'll need to add them to your git repository in any case.

**Reply by saywebsolutions** [reply]
Thanks I got it working by seeing what the install command did and executing the steps manually, everything seems to be working except all assets are 404ing, does Twill use the laravel `asset()` helper function like the vapor docs recommend? I verified the assets got published successfully with the

```
vendor:publish --provider='A17\Twill\TwillServiceProvider' --tag=assets
```

command

**Reply by saywebsolutions** [reply]
I ended up publishing the views and wrapping the assets in the `asset()` helper and that seems to be working

**Comment by antonioribeiro** [comment]
No need to publish the views, you can just overload the twillAsset() helper, that’s what we are doing

**Reply by saywebsolutions** [reply]
thanks, so what's the best spot to overload twillAsset() (where should I put my override in my laravel project)?

and to "unpublish" the views can i just delete the `resources/views/vendor/twill/` directory and it will revert back to using the ones supplied with the package?

Thanks again for the help

**Comment by antonioribeiro** [comment]
Well, because Composer may autoload package files before hitting the
application, I’m loading my helpers files “manually” before the require
autoload.php. There’s probably a better way of doing that, but I wasn’t
able to do it better

On Wed, 28 Jul 2021 at 21:25 saywebsolutions ***@***.***>
wrote:

> thanks, so what's the best spot to overload twillAsset() (where should I
> put my override in my laravel project)?
>
> and to "unpublish" the views can i just delete the
> resources/views/vendor/twill/ directory and it will revert back to using
> the ones supplied with the package?
>
> Thanks again for the help
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/area17/twill/discussions/1065#discussioncomment-1083969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYJCEDVKDXLTCNM2GEN6Q3T2BKTRANCNFSM5A67I4RQ>
> .
>
-- 
Antonio Carlos Ribeiro
***@***.***

Enviado do meu iPhone
